dts/bindings/sensor/st,stm32-qdec.yaml

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2022, Valerio Setti <vsetti@baylibre.com
+# SPDX-License-Identifier: Apache-2.0
+
+description: STM32 quadrature decoder
+
+compatible: "st,stm32-qdec"
+
+include:
+    - name: base.yaml
+    - name: pinctrl-device.yaml
+
+properties:
+     pinctrl-0:
+       required: true
+
+     pinctrl-names:
+       required: true
+
+     st,input-polarity-inverted:
+       type: boolean
+       required: false
+       description: Encoder is triggered by a falling edge on the input pin
+
+     st,input-filter-level:
+       type: int
+       required: true
+       description: |
+         Intensity of the filter applied to the input signal. This is
+         implemented by scaling the sampling frequency and adding a counter
+         in which N consecutive samples with same value are needed to validate
+         a transition.
+         Mapping is as follows (F_clk is the timer's clock):
+         0: No filter, sampling is done at F_dts (default value)
```

Comment:
It does not make sense to define a default value and mark the property as required.
Please choose either or.